docs/source/cli-reference/core-options.rst

```diff
@@ -119,3 +119,25 @@ Comparing Progress Message Options
     An example usage::
 
         scancode -clieu --json-pp sample.json samples --max-in-memory -1
+
+----
+
+``--max_depth`` Option
+----------------------------------
+
+    Normally, the scan takes place upto the maximum level of nesting of directories possible. But
+    using the ``--max-depth`` option, the depth of the scan can be limited. This can reduce the time
+    taken for the scan when deeper directories are not relevant.
+
+    Note that the ``--max-depth`` option will be ignored if you are scanning from a JSON file using
+    the ``--from-json`` option. In that case, the original depth is used.
+
+    Some INTEGER values of the ``--max-depth INTEGER`` option:
+
+    - **4**     - Scan only upto 4 levels of subdirectories
```

Comment:
Say I scan ~/foo/  would this mean ~/foo/one/two/three/four/file5 would be scanned? or that ~/foo/one/two/three/file4 would be scanned?

Reply:
With a depth of 4, that would scan upto `file4`. The contents of `foo` will be treated as having a depth of 1. Thus the contents of `three` are having a depth of 4. We would not recurse any deeper than that.

Reply:
Thank you fr the explanantion

Reply:
Yeah, I see. Would something like this be appropriate?
*Descend at most INTEGER levels of directories below the starting-point.  A zero or negative value indicates no limit.*
This is similar wording to what's on the manpage of `find` utility.

Reply:
What about this: `Descend at most INTEGER levels of directories below and including the starting-point. Use zero for no limit.` ?